### PR TITLE
Compendium client has polymorphic effect type. 

### PR DESCRIPTION
**Pull request by rlmark**
Apologies for the delay, As can be seen from the commit log, getting this branch to build in TravisCI was an adventure. This PR bumps the SBT version to 1.3.4. The actual code removes the hard-coded IO effect type in favor of a polymorphic effect type, addressing issue: https://github.com/higherkindness/compendium/issues/114. 

One thing to note: I wasn't sure wether to bump the version from 0.0.1-SNAPSHOT to 0.0.2-SNAPSHOT here, or if that would be auto-incremented via a build process. I definitely don't want to clobber any artifacts we might be relying on in other projects. If you'd like me to bump the patch (or other) version in this PR, I'm happy to do so!

### CODE REVIEW

**Comment by pirita** [comment]
AFAIK, We don't have auto-incremented version in compendium and I think we should bump the version to be able to differentiate the polymorphic version from the non-polymorphic version when updating the plugin.

File: version.sbt [hunk 1]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.0.1-SNAPSHOT"
+version in ThisBuild := "0.0.2-SNAPSHOT"
```
Comment (JesusMtnez) [inline version.sbt]: Right now there is no official release of `compendium`. IMHO, I don't think it is necessary to bump the version right now. Once `0.0.1` is released, `sbt-org-policies` will set the version to `0.0.2-SNAPSHOT` as its next action.

I think `version` should only be changed to stable versions so the CI makes a release. But it is just my opinion, of course :smile:

File: modules/client/src/main/scala/higherkindness/compendium/CompendiumClient.scala [hunk 2]
```diff
@@ -16,99 +16,104 @@
...
+          out <- {
+            val statusToF: F[Option[Protocol]] = status match {
+              case Status.OK       => request.as[Protocol].map(Option(_)).exec[F]
+              case Status.NotFound => Sync[F].pure(None)
+              case Status.InternalServerError =>
+                Sync[F].raiseError(UnknownError(s"Error in compendium server"))
+              case _ =>
+                Sync[F].raiseError(UnknownError(s"Unknown error with status code $status"))
+            }
+            statusToF
```
Comment (JesusMtnez) [inline modules/client/src/main/scala/higherkindness/compendium/CompendiumClient.scala]: Just out of curiosity, why creating `statusToF` to return it later?

Comment (rlmark) [reply modules/client/src/main/scala/higherkindness/compendium/CompendiumClient.scala]: Ah, good question, I initially tried without the `val` and it seems the Scala compiler can't interpret the case statement as total on the right side of the for expression without a little bit of type help. I'm open to ideas here, as I would have preferred not to create the value only to return it.

File: version.sbt (hunk 1 above)
Comment (rlmark) [reply version.sbt]: Ok, sounds good. I will downgrade back to to 0.0.1-SNAPSHOT.

File: modules/client/src/main/scala/higherkindness/compendium/CompendiumClient.scala (hunk 2 above)
Comment (JesusMtnez) [reply modules/client/src/main/scala/higherkindness/compendium/CompendiumClient.scala]: Nice! Thank you @rlmark !!

**Comment by JesusMtnez** [comment]
Now that the client has been updated, we should update this PR with the new client implementation: https://github.com/higherkindness/sbt-compendium/pull/4

**Comment by rlmark** [comment]
> Now that the client has been updated, we should update this PR with the new client implementation: [higherkindness/sbt-compendium#4](https://github.com/higherkindness/sbt-compendium/pull/4)

Awesome, will take a look at that next!

**Comment by rlmark** [comment]
> Now that the client has been updated, we should update this PR with the new client implementation: [higherkindness/sbt-compendium#4](https://github.com/higherkindness/sbt-compendium/pull/4)

Just for record-keeping, this is change in the client is integrated in sbt-compendium on [this branch](https://github.com/higherkindness/sbt-compendium/tree/ign/compendium-plugin)
with (this commit)[https://github.com/higherkindness/sbt-compendium/tree/ign/compendium-plugin]

**Pull request merged**